Index: lib/Hero/page1.dart
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/lib/Hero/page1.dart b/lib/Hero/page1.dart
new file mode 100644
--- /dev/null	(date 1677227498989)
+++ b/lib/Hero/page1.dart	(date 1677227498989)
@@ -0,0 +1,50 @@
+import 'package:flutter/material.dart';
+import 'package:practical_5/Hero/page2.dart';
+
+class HeroPage1 extends StatelessWidget {
+  const HeroPage1({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Hero Animation'),
+      ),
+      body: Row(
+        children: [
+          GestureDetector(
+            onTap: () {
+              Navigator.push(
+                context,
+                PageRouteBuilder(
+                  transitionDuration: const Duration(seconds: 5),
+                  pageBuilder: (_, __, ___) => const HeroPage2(),
+                ),
+              );
+            },
+            child: Hero(
+              tag: 'Hero',
+              transitionOnUserGestures: true,
+              flightShuttleBuilder: (flightContext, animation, flightDirection,
+                  fromHeroContext, toHeroContext) {
+                if (flightDirection == HeroFlightDirection.push) {
+                  return const Icon(Icons.push_pin);
+                } else {
+                  return const Icon(Icons.remove);
+                }
+              },
+              child: const Icon(
+                Icons.add_circle,
+                size: 100,
+              ),
+            ),
+          ),
+          const Icon(
+            Icons.remember_me,
+            size: 100,
+          ),
+        ],
+      ),
+    );
+  }
+}
Index: lib/Hero/page2.dart
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/lib/Hero/page2.dart b/lib/Hero/page2.dart
new file mode 100644
--- /dev/null	(date 1677226216013)
+++ b/lib/Hero/page2.dart	(date 1677226216013)
@@ -0,0 +1,41 @@
+import 'package:flutter/material.dart';
+
+class HeroPage2 extends StatelessWidget {
+  const HeroPage2({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Hero Animation Page2'),
+      ),
+      body: Hero(
+        tag: 'Hero',
+        child: const Center(
+          child: Icon(
+            Icons.add,
+            size: 200,
+          ),
+        ),
+        flightShuttleBuilder: (flightContext, animation, flightDirection,
+            fromHeroContext, toHeroContext) {
+          if (flightDirection == HeroFlightDirection.push) {
+            return const Icon(Icons.push_pin);
+          } else {
+            return const Icon(Icons.remove);
+          }
+        },
+        transitionOnUserGestures: true,
+        placeholderBuilder: (context, heroSize, child) {
+          return Center(
+            child: Container(
+              height: 300,
+              width: 300,
+              child: const CircularProgressIndicator(),
+            ),
+          );
+        },
+      ),
+    );
+  }
+}
Index: lib/main.dart
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/lib/main.dart b/lib/main.dart
--- a/lib/main.dart	(revision bb1fe3a5ee85fa4336e6250cfb73b65800f41b20)
+++ b/lib/main.dart	(date 1677217304243)
@@ -1,5 +1,5 @@
 import 'package:flutter/material.dart';
-import 'package:practical_5/AdvancedWidget/wrap_widget.dart';
+import 'package:practical_5/Hero/page1.dart';
 
 void main() {
   // TODO: implement main
@@ -11,8 +11,11 @@
 
   @override
   Widget build(BuildContext context) {
-    return const MaterialApp(
-      home: WrapWidget(),
+    return MaterialApp(
+      theme: ThemeData(
+        useMaterial3: true,
+      ),
+      home: HeroPage1(),
       //debugShowCheckedModeBanner: false,
     );
   }
Index: lib/AdvancedWidget/wrap_widget.dart
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/lib/AdvancedWidget/wrap_widget.dart b/lib/AdvancedWidget/wrap_widget.dart
--- a/lib/AdvancedWidget/wrap_widget.dart	(revision bb1fe3a5ee85fa4336e6250cfb73b65800f41b20)
+++ b/lib/AdvancedWidget/wrap_widget.dart	(date 1676365544145)
@@ -1,144 +1,144 @@
-import 'package:flutter/material.dart';
-
-class WrapWidget extends StatelessWidget {
-  const WrapWidget({Key? key}) : super(key: key);
-
-  @override
-  Widget build(BuildContext context) {
-    return Scaffold(
-      appBar: AppBar(
-        title: Text("WRAP WIDGET"),
-        centerTitle: true,
-        backgroundColor: Colors.deepPurple,
-      ),
-      body: Column(
-        children: [
-          // Wrap(
-          //   direction: Axis.horizontal,
-          //   textDirection: TextDirection.rtl,
-          //   verticalDirection: VerticalDirection.up,
-          //   children: [
-          //     Container(
-          //       height: 50,
-          //       width: 50,
-          //       color: Colors.black,
-          //     ),
-          //     SizedBox(
-          //       width: 10,
-          //     ),
-          //     Container(
-          //       height: 50,
-          //       width: 50,
-          //       color: Colors.deepPurple,
-          //     ),
-          //     SizedBox(
-          //       width: 10,
-          //     ),
-          //     Container(
-          //       height: 50,
-          //       width: 50,
-          //       color: Colors.teal,
-          //     ),
-          //     SizedBox(
-          //       width: 10,
-          //     ),
-          //     Container(
-          //       height: 50,
-          //       width: 50,
-          //       color: Colors.purpleAccent,
-          //     ),
-          //     SizedBox(
-          //       width: 10,
-          //     ),
-          //     Container(
-          //       height: 50,
-          //       width: 50,
-          //       color: Colors.yellowAccent,
-          //     ),
-          //     SizedBox(
-          //       width: 10,
-          //     ),
-          //     Container(
-          //       height: 50,
-          //       width: 50,
-          //       color: Colors.redAccent,
-          //     ),
-          //     SizedBox(
-          //       width: 10,
-          //     ),
-          //     Container(
-          //       height: 50,
-          //       width: 50,
-          //       color: Colors.lightGreen,
-          //     ),
-          //     SizedBox(
-          //       width: 10,
-          //     ),
-          //     Container(
-          //       height: 50,
-          //       width: 50,
-          //       color: Colors.limeAccent,
-          //     ),
-          //   ],
-          // ),
-          InkWell(
-            onTap: () {
-              print("Inkwell Pressed");
-            },
-            child: Icon(
-              Icons.fingerprint_sharp,
-              size: 50,
-            ),
-          ),
-          Divider(
-            indent: 15,
-            endIndent: 15,
-            thickness: 5,
-          ),
-          ElevatedButton(
-            onPressed: () {},
-            child: GestureDetector(
-                onTap: () {
-                  print("Gesture Detector Calle");
-                },
-                child: Text("Hi, How Are You?")),
-          ),
-          SizedBox(
-            height: 10,
-          ),
-          // ClipOval(
-          //     child: Image(
-          //   width: 150,
-          //   height: 150,
-          //   image: NetworkImage('https://i.ibb.co/1vXpqVs/flutter-logo.jpg'),
-          // )),
-          ClipRRect(
-            child: ClipPath(
-              clipper: MyCustomClipper(),
-              child: Image.network("https://i.ibb.co/1vXpqVs/flutter-logo.jpg"),
-            ),
-          ),
-        ],
-      ),
-    );
-  }
-}
-
-class MyCustomClipper extends CustomClipper<Path> {
-  @override
-  getClip(Size size) {
-    // TODO: implement getClip
-    final path = Path();
-    path.lineTo(size.width, 0.0);
-    path.lineTo(size.width * 0.75, size.height);
-    path.lineTo(size.width * 0.25, size.height);
-    path.close();
-    return path;
-  }
-
-  @override
-  bool shouldReclip(covariant CustomClipper oldClipper) {
-    // TODO: implement shouldReclip
-    throw UnimplementedError();
-  }
-}
+// import 'package:flutter/material.dart';
+//
+// class WrapWidget extends StatelessWidget {
+//   const WrapWidget({Key? key}) : super(key: key);
+//
+//   @override
+//   Widget build(BuildContext context) {
+//     return Scaffold(
+//       appBar: AppBar(
+//         title: Text("WRAP WIDGET"),
+//         centerTitle: true,
+//         backgroundColor: Colors.deepPurple,
+//       ),
+//       body: Column(
+//         children: [
+//           Wrap(
+//             direction: Axis.horizontal,
+//             textDirection: TextDirection.rtl,
+//             verticalDirection: VerticalDirection.up,
+//             children: [
+//               Container(
+//                 height: 50,
+//                 width: 50,
+//                 color: Colors.black,
+//               ),
+//               SizedBox(
+//                 width: 10,
+//               ),
+//               Container(
+//                 height: 50,
+//                 width: 50,
+//                 color: Colors.deepPurple,
+//               ),
+//               SizedBox(
+//                 width: 10,
+//               ),
+//               Container(
+//                 height: 50,
+//                 width: 50,
+//                 color: Colors.teal,
+//               ),
+//               SizedBox(
+//                 width: 10,
+//               ),
+//               Container(
+//                 height: 50,
+//                 width: 50,
+//                 color: Colors.purpleAccent,
+//               ),
+//               SizedBox(
+//                 width: 10,
+//               ),
+//               Container(
+//                 height: 50,
+//                 width: 50,
+//                 color: Colors.yellowAccent,
+//               ),
+//               SizedBox(
+//                 width: 10,
+//               ),
+//               Container(
+//                 height: 50,
+//                 width: 50,
+//                 color: Colors.redAccent,
+//               ),
+//               SizedBox(
+//                 width: 10,
+//               ),
+//               Container(
+//                 height: 50,
+//                 width: 50,
+//                 color: Colors.lightGreen,
+//               ),
+//               SizedBox(
+//                 width: 10,
+//               ),
+//               Container(
+//                 height: 50,
+//                 width: 50,
+//                 color: Colors.limeAccent,
+//               ),
+//             ],
+//           ),
+//           InkWell(
+//             onTap: () {
+//               print("Inkwell Pressed");
+//             },
+//             child: Icon(
+//               Icons.fingerprint_sharp,
+//               size: 50,
+//             ),
+//           ),
+//           Divider(
+//             indent: 15,
+//             endIndent: 15,
+//             thickness: 5,
+//           ),
+//           ElevatedButton(
+//             onPressed: () {},
+//             child: GestureDetector(
+//                 onTap: () {
+//                   print("Gesture Detector Calle");
+//                 },
+//                 child: Text("Hi, How Are You?")),
+//           ),
+//           SizedBox(
+//             height: 10,
+//           ),
+//           // ClipOval(
+//           //     child: Image(
+//           //   width: 150,
+//           //   height: 150,
+//           //   image: NetworkImage('https://i.ibb.co/1vXpqVs/flutter-logo.jpg'),
+//           // )),
+//           ClipRRect(
+//             child: ClipPath(
+//               clipper: MyCustomClipper(),
+//               child: Image.network("https://i.ibb.co/1vXpqVs/flutter-logo.jpg"),
+//             ),
+//           ),
+//         ],
+//       ),
+//     );
+//   }
+// }
+//
+// class MyCustomClipper extends CustomClipper<Path> {
+//   @override
+//   getClip(Size size) {
+//     // TODO: implement getClip
+//     final path = Path();
+//     path.lineTo(size.width, 0.0);
+//     path.lineTo(size.width * 0.75, size.height);
+//     path.lineTo(size.width * 0.25, size.height);
+//     path.close();
+//     return path;
+//   }
+//
+//   @override
+//   bool shouldReclip(covariant CustomClipper oldClipper) {
+//     // TODO: implement shouldReclip
+//     throw UnimplementedError();
+//   }
+// }
Index: lib/AdvancedWidget/page_view.dart
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/lib/AdvancedWidget/page_view.dart b/lib/AdvancedWidget/page_view.dart
new file mode 100644
--- /dev/null	(date 1677217212747)
+++ b/lib/AdvancedWidget/page_view.dart	(date 1677217212747)
@@ -0,0 +1,51 @@
+import 'package:flutter/material.dart';
+
+class PageViewWidget extends StatefulWidget {
+  const PageViewWidget({Key? key}) : super(key: key);
+
+  @override
+  State<PageViewWidget> createState() => _PageViewWidgetState();
+}
+
+class _PageViewWidgetState extends State<PageViewWidget> {
+  PageController controller = PageController();
+  int _curr = 0;
+
+  @override
+  Widget build(BuildContext context) {
+    return Theme(
+      data: ThemeData(
+        useMaterial3: true,
+      ),
+      child: Scaffold(
+        appBar: AppBar(
+          title: const Text("PAGE VIEW "),
+          centerTitle: true,
+        ),
+        body: PageView(
+          allowImplicitScrolling: true,
+          scrollDirection: Axis.vertical,
+          controller: controller,
+          scrollBehavior: MaterialScrollBehavior(),
+          children: const [
+            Center(
+              child: Text("Page One"),
+            ),
+            Center(
+              child: Text("Page 2"),
+            ),
+            Center(
+              child: Text("Page 3"),
+            ),
+          ],
+          onPageChanged: (value) {
+            setState(() {
+              print(value);
+              _curr = value;
+            });
+          },
+        ),
+      ),
+    );
+  }
+}
